migration/1716549958362-add_donations_mannually_to_db.ts

```diff
@@ -0,0 +1,285 @@
+    valueUsd: 33.86,
+    transactionNetworkId: NETWORK_IDS.MAIN_NET,
+    createdAt: millisecondTimestampToDate(1715201051000),
+  },
+];
+
+export class AddDonationsMannuallyToDb1716549958362
+  implements MigrationInterface
+{
+  async up(queryRunner: QueryRunner): Promise<void> {
+    const environment = config.get('ENVIRONMENT') as string;
+
+    if (environment !== 'production') {
+      // eslint-disable-next-line no-console
+      console.log('We just want to create these donations in production DB');
+      return;
+    }
+
+    await AppDataSource.initialize();
+    for (const tx of transactions) {
+      let user = (
+        await queryRunner.query(`SELECT * FROM public.user
+        WHERE lower("walletAddress")=lower('${tx.donorAddress}')`)
+      )[0];
+      if (!user) {
+        // eslint-disable-next-line no-console
+        console.log('User is not in our DB, creating .... ');
+        await queryRunner.query(`
+                    INSERT INTO public.user ("walletAddress", role,"loginType", name) 
+                    VALUES('${tx?.donorAddress?.toLowerCase()}', 'restricted','wallet', '${tx.donorName}');
+                    `);
+        user = (
+          await queryRunner.query(`SELECT * FROM public.user
+                        WHERE lower("walletAddress")=lower('${tx.donorAddress}')`)
+        )[0];
+      }
+
+      // Set true for isTokenEligibleForGivback, isProjectVerified because Ashley mentioned we want to pay givback for them
+      const createdAt = moment(tx.createdAt).format('YYYY-MM-DD HH:mm:ss');
+      const project = (await findProjectById(
+        tx.projectId as number,
+      )) as Project;
+
+      const { givbackFactor, projectRank, powerRound, bottomRankInRound } =
+        await calculateGivbackFactor(tx.projectId as number);
+
+      await queryRunner.query(`
+           INSERT INTO donation ("toWalletAddress", "projectId", "fromWalletAddress", "userId", amount, currency, "transactionId", "transactionNetworkId", anonymous, "valueUsd", status,
+            "segmentNotified", "isTokenEligibleForGivback", "isProjectVerified", "createdAt", "givbackFactor", "powerRound", "projectRank", "bottomRankInRound", "qfRoundId", "tokenAddress")
+           VALUES ('${tx.toWalletAddress?.toLowerCase()}', ${
+             tx.projectId
+           }, '${tx.fromWalletAddress?.toLocaleLowerCase()}', ${user.id}, ${
+             tx.amount
+           }, '${tx.currency}', '${tx.transactionId?.toLocaleLowerCase()}', ${
+             tx.transactionNetworkId
+           }, false, ${tx.valueUsd}, 'verified',
+             true, true, true, '${createdAt}', ${givbackFactor}, ${powerRound}, ${projectRank}, ${bottomRankInRound}, ${tx.qfRoundId || null}, '${
+               tx.tokenAddress
+             }');
+                `);
+
+      await updateUserTotalDonated(user.id);
+      await updateUserTotalReceived(project.adminUser?.id);
+      await updateTotalDonationsOfProject(tx.projectId as number);
+    }
+
```

Comment:
I added the method updateProjectStatistics() for the project_donation_summary_view, maybe we have to call it here too. 

The rest looks good.

Reply:
Might need to rebase master into this branch

Reply:
Thanks @CarlosQ96